icu4c/source/i18n/plurrule.cpp

```diff
@@ -707,54 +746,62 @@ PluralRules::getRuleFromResource(const Locale& locale, UPluralType type, UErrorC
 UnicodeString
 PluralRules::getRules() const {
     UnicodeString rules;
-    if (mRules != NULL) {
+    if (mRules != nullptr) {
         mRules->dumpRules(rules);
     }
     return rules;
 }
 
-
 AndConstraint::AndConstraint() {
     op = AndConstraint::NONE;
     opNum=-1;
     value = -1;
-    rangeList = NULL;
+    rangeList = nullptr;
     negated = FALSE;
     integerOnly = FALSE;
     digitsType = none;
-    next=NULL;
+    next = nullptr;
+    fInternalStatus = U_ZERO_ERROR;
 }
 
-
 AndConstraint::AndConstraint(const AndConstraint& other) {
+    this->fInternalStatus = other.fInternalStatus;
+    if (U_FAILURE(fInternalStatus)) {
```

Comment:
Need to get the object fully initialized, esp. `rangeList` and `next` before returning, so that the destructor will not try to delete uninitialized pointers. With C++11, we could just move the initialization from the default constructor into the class declaration, using direct member initialization. Then this constructor could stay as it is.

Reply:
Thank you again Andy!
I think I will use the direct member initialization in the class declaration for this so then the default constructor can go away.